enhancements/sig-architecture/105-aws-iam-registration/README.md

```diff
@@ -0,0 +1,464 @@
+            }
+         }
+      },
+      {
+         "Effect": "Allow",
+         "Principal": {
+            "Federated": "arn:aws:iam::<managed-cluster-account-id>:oidc-provider/<oidc-provider>"
+         },
+         "Action": "sts:AssumeRoleWithWebIdentity",
+         "Condition": {
+            "StringEquals": {
+               "<oidc-provider>:sub": "system:serviceaccount:open-cluster-management-agent:<klusterlet-name>-work-sa",
+               "<oidc-provider>:aud": "sts.amazonaws.com"
+            }
+         }
+      }
+   ]
+}
+```
+
+With following tags:
+```
+hub_account_id: <hub-account-id>
+hub_cluster_name: <hub-cluster-name>
+managed_cluster_name: <managed-cluster-name>
+```
+
+**Note:** The _md5-hash-unique-identifier_ should be calculated, using the hash of the values of tags (`hub_account_id`, `hub_cluster_name`, and `managed_cluster_name`) on the role concatenated using `#` as the delimiter, in the alphabetical order of their keys:
+
+For example, given the following tags:
+- `hub_account_id: hubaccount1`
+- `hub_cluster_name: hubcluster1`
+- `managed_cluster_name: spokecluster1`
+
+The md5-hash-unique-identifier should be `hash(hubaccount1#hubcluster1#spokecluster1)` which is `1c502b1d59797fea758ae8177f246b2f`.
+
+#### Hub Cluster Prerequisites
+
+An IAM role named using following template:
+```
+arn:aws:iam::<hub-account-id>:role/<hub-cluster-name>_managed-cluster-identity-creator
+```
+
+Having the following access permissions policy:
+```json
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": [
+        "iam:CreateRole",
+        "iam:DeleteRole",
+        "iam:UpdateAssumeRolePolicy",
+        "iam:PutRolePolicy",
+        "iam:DeleteRolePolicy",
+        "iam:AttachRolePolicy",
+        "iam:DetachRolePolicy",
+        "iam:CreatePolicy",
+        "iam:DeletePolicy",
+        "iam:CreatePolicyVersion",
+        "iam:DeletePolicyVersion",
+        "iam:ListAttachedRolePolicies",
+        "iam:ListRolePolicies",
+        "iam:ListPolicyVersions",
+        "iam:GetRole",
+        "iam:GetRolePolicy",
+        "iam:GetPolicy",
+        "iam:GetPolicyVersion",
+        "iam:PassRole"
+      ],
+      "Resource": [
+        "arn:aws:iam::<hub-account-id>:role/ocm-hub-*",
+        "arn:aws:iam::<hub-account-id>:policy/ocm-hub-*"
+      ]
+    }
+  ]
+}
+```
+
+And following trust relationship policy:
+```json
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "Federated": "arn:aws:iam::<hub-account-id>:oidc-provider/<oidc-provider>"
+      },
+      "Action": "sts:AssumeRoleWithWebIdentity",
+      "Condition": {
+        "StringEquals": {
+          "<oidc-provider>:sub": "system:serviceaccount:open-cluster-management-hub:registration-controller-sa",
+          "<oidc-provider>:aud": "sts.amazonaws.com"
+        }
+      }
+    }
+  ]
+}
+```
+
+#### Cluster join initiated from managed cluster
```

Comment:
We also have some addons that need to access the Hub cluster apiserver, it would be cool if the addon registration workflow was shown here as well.

Reply:
@zhujian7 we do not have very good understanding of the addon framework at this point and couldn't think about the finer details of addon registration process. Having said that, my understanding is that if we use the registration driver approach proposed by @qiujian16 the addon registration should be similar to core registration. 

I would request your help on this, if you agree with the core registration approach proposed here. Could you please help extend the same to addon registration?

Reply:
yes, I think we could add addon registration as non-goal in this proposal, and try to do it in another one. This will limit this for only cluster registration, but should be fine as an alpha feature.